File: backend/src/main/java/com/example/sneakers/model/Sneaker.java
package com.example.sneakers.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.URL;

import javax.persistence.*;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.math.BigDecimal;
import java.util.Date;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Sneaker {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long Id;

    @NotBlank
    @Size(min = 3, max = 100)
    private String brand;

    @NotBlank
    @Size(min = 3, max = 100)
    private String name;

    @Lob
    @Size(min = 3, max = 300)
    private String description;

    @DecimalMin(value = "0.0", inclusive = false)
    private BigDecimal price;

    @URL(protocol = "http")
    @NotBlank
    private String imageUrl;

    @URL
    @NotBlank
    private String shopUrl;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date releaseDate;

}

File: frontend/src/app/model/sneaker.ts
export class Sneaker {
  id: Number;
  brand: String;
  name: String;
  price: Number;
  imageUrl: String;
  shopUrl: String;
  releaseDate: Date;
  description: String;

  constructor(id: Number, brand: String, description: String, name: String, price: Number, imageUrl: String, shopUrl: String, releaseDate: Date) {
    this.id = id;
    this.brand = brand;
    this.name = name;
    this.price = price;
    this.imageUrl = imageUrl;
    this.shopUrl = shopUrl;
    this.releaseDate = releaseDate;
    this.description = description;
  }
}
